ps1/ps1/src/twitter/Extract.java

```diff
@@ -1,177 +0,0 @@
-/* Copyright (c) 2007-2016 MIT 6.005 course staff, all rights reserved.
- * Redistribution of original or derived work requires permission of course staff.
- */
-package twitter;
-
-import java.util.List;
-import java.util.Set;
-import java.util.HashSet;
-import java.time.Instant;
-import twitter.Timespan;
-import java.lang.StringBuilder;
-
-
-/**
- * Extract consists of methods that extract information from a list of tweets.
- * 
- * DO NOT change the method signatures and specifications of these methods, but
- * you should implement their method bodies, and you may add new public or
- * private methods or classes if you like.
- */
-public class Extract {
-
-    /**
-     * Get the time period spanned by tweets.
-     * 
-     * @param tweets
-     *            list of tweets with distinct ids, not modified by this method.
-     * @return a minimum-length time interval that contains the timestamp of
-     *         every tweet in the list.
-     */
-    public static Timespan getTimespan(List<Tweet> tweets) {
-        
-        if (tweets.size() < 1) {
-            throw new RuntimeException("Tweet list doesn't fit spec.");
-        }
-        
-        // set the starting min/max
-        Instant min = tweets.get(0).getTimestamp();
-        Instant max = tweets.get(0).getTimestamp();
-        
-        // replace min and max time if min-er or max-er time is found
-        for (int i = 1; i < tweets.size(); i++){
-            
-            Instant current = tweets.get(i).getTimestamp();
-            if (current.isAfter(max)) {
-                max = current;
-                }
-            if (current.isBefore(min)) {
-                min = current;
-                }
-        }
-                
-        Timespan span = new twitter.Timespan(min, max);
-                        
-        return span;
-        
-    }
-    
-    /**
-     * Find and return the next username from position at_char in a String text.
-     * 
-     * @param text of a tweet (string)
-     *        at_char is the index from which to search the next username
-     * @return the next username, converted to lowercase String. If none, returns
-     * an empty String.
-     */
-    public static String getUsername(String text, int at_char) {
-        
-        String AcceptableChars = 
-                "abcdefghijklmnopqrstuvwxyz"
-              + "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
-              + "0123456789"
-              + "-_" ;
-        
-        char char_preceding;
-        char char_succeeding;
-
-        if(at_char == -1){
-            throw new RuntimeException("@ symbol doesn't appear in tweet.");
```

Comment:
Yes you are completely right. It doesn't make sense that this method would throw that exception if you pass `-1`, it should only throw a generic error that the index is outside the scope of the tweet. 

I think in my main method implementation , it can pop out... so I should have added an if statement in the main method like `if(at_char = -1) {continue} else {getUsername(....)}`.